nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/autodiff/samediff/ops/SDNN.java

```diff
@@ -1032,4 +1032,35 @@ public SDVariable multiHeadDotProductAttention(String name, SDVariable queries,
             );
         }
     }
+
+    /**
+     * Max pooling on the input and outputs both max values and indices
+     *
+     * @param name  Name of the output variable
+     * @param x
+     * @return output array and argmax array
+     */
+    public SDVariable[] maxPoolWithArgmaxs(String name, SDVariable x) {
+        SDVariable[] res = f().maxPoolWithArgmaxs(x);
+        return res;
```

Comment:
missing updateVariableNameAndReference
Because there are 2 outputs, let's use this:
```
if(name != null){
    updateVariableNameAndReference(res[0], name + "/output");
    updateVariableNameAndReference(res[1], name + "/index");
}
```

Reply:
Also `@NonNull` and javadoc tag for pooling2DConfig arg